testing/src/test/java/org/motechproject/nms/testing/it/api/UserControllerBundleIT.java

```diff
@@ -1479,5 +1496,538 @@ public void verifyFT184() throws IOException,
                 EntityUtils.toString(response.getEntity()));
     }
 
-    
+    private void setupWhiteListData() {
+
+	rh = new RegionHelper(languageDataService, circleDataService,
+		stateDataService, districtDataService, districtService);
+	rh.newDelhiDistrict();
+	rh.bangaloreDistrict();
+
+	whitelistState = rh.delhiState();
+	stateDataService.create(whitelistState);
+
+	whitelistStateDataService.create(new WhitelistState(whitelistState));
+
+	nonWhitelistState = rh.karnatakaState();
+	stateDataService.create(nonWhitelistState);
+    }
+
+    HttpGet createHttpGetUserDetails(Service serviceType, String callingNo,
+	    String operator, String circle, String callId) {
+	StringBuilder sb = new StringBuilder();
+	String endpoint = "";
+	if (serviceType == Service.MOBILE_ACADEMY) {
+	    endpoint = String.format(
+		    "http://localhost:%d/api/mobileacademy/user",
+		    TestContext.getJettyPort());
+	} else if (serviceType == Service.MOBILE_KUNJI) {
+	    endpoint = String.format(
+		    "http://localhost:%d/api/mobilekunji/user",
+		    TestContext.getJettyPort());
+	}
+	sb.append(endpoint);
+	String seperator = "?";
+	if (callingNo != null) {
+	    sb.append(seperator);
+	    sb.append("callingNumber=");
+	    sb.append(callingNo);
+	    seperator = "";
+	}
+	if (operator != null) {
+	    if (seperator.equals("")) {
+		sb.append("&");
+	    } else {
+		sb.append(seperator);
+	    }
+	    sb.append("operator=");
+	    sb.append(operator);
+	}
+	if (circle != null) {
+	    if (seperator.equals("")) {
+		sb.append("&");
+	    } else {
+		sb.append(seperator);
+	    }
+	    sb.append("circle=");
+	    sb.append(circle);
+	}
+	if (callId != null) {
+	    if (seperator.equals("")) {
+		sb.append("&");
+	    } else {
+		sb.append(seperator);
+	    }
+	    sb.append("callId=");
+	    sb.append(callId);
+	}
+	// System.out.println("Request url:" + sb.toString());
+	return RequestBuilder.createGetRequest(sb.toString());
+    }
+
+    HttpPost createHttpPostSetLanguageLocationCode(Service serviceType,
```

Comment:
Sorry, I was not aware of the methods. I have changed the methods and committed the code as per your instructions.